ide/diff/src/org/netbeans/modules/diff/builtin/visualizer/editable/EditableDiffView.java

```diff
@@ -372,6 +392,38 @@ public void run () {
         }
     }
 
+    private static String detectLineEnding(StreamSource source) {
+        try {
+            FileObject fo = source.getLookup().lookup(FileObject.class);
+            DataObject dao = DataObject.find(fo);
+
+            EditorCookie editorCookie = dao.getCookie(EditorCookie.class);
+            if (editorCookie == null) {
+                return null;
+            }
+
+            Document doc;
+            try {
+                doc = editorCookie.openDocument();
+            } catch (IOException ex) {
+                return null;
+            }
```

Comment:
This will fail if the source is not backed by a file. If you pass in the `DiffContentPanel`, you can simplify this to:


```suggestion
    private static String detectLineEnding(DiffContentPanel source) {
        try {
            Document doc = source.getEditorPane().getDocument();
```

The panels are directly available in `RefreshDiffTask#diff`.